lib/full-site-editing/edit-site-page.php

```diff
@@ -248,3 +249,27 @@ function gutenberg_site_editor_load_block_editor_scripts_and_styles( $is_block_e
 		: $is_block_editor_screen;
 }
 add_filter( 'should_load_block_editor_scripts_and_styles', 'gutenberg_site_editor_load_block_editor_scripts_and_styles' );
+
+/**
+ * Do a server-side redirection if missing `postType` and `postId`
+ * query args when visiting site editor.
+ *
+ * Note: The `site-editor.php` should handle redirection when migrated into the WP core.
+ *
+ * @return void
+ */
+function gutenberg_maybe_redirect_to_homepage() {
+	if ( empty( $_GET['postType'] ) && empty( $_GET['postId'] ) ) {
+		$template = gutenberg_resolve_home_template();
+		if ( ! $template ) {
+			return;
+		}
+
+		$redirect_url = add_query_arg(
+			$template,
+			admin_url( 'themes.php?page=gutenberg-edit-site' )
+		);
+		wp_safe_redirect( $redirect_url );
```

Comment:
It looks like whenever we find a template with this function, we are redirecting.

Is there a case where the template this function returns is the same as the template the editor initially resolved in the first place, causing the redirect to be unnecessary?

Reply:
> Is there a case where the template this function returns is the same as the template the editor would initially resolve in the first place, causing the redirect to be unnecessary?

I don't think so. Site Editor needs these query args to resolve the template.